pandas/core/window/rolling.py

```diff
@@ -235,10 +237,13 @@ def _check_window_bounds(
                 f"start ({len(start)}) and end ({len(end)}) bounds must be the "
                 f"same length"
             )
-        elif len(start) != num_vals:
+        elif not isinstance(self._get_window_indexer(), GroupbyIndexer) and len(
+            start
+        ) != (num_vals + (self.step or 1) - 1) // (self.step or 1):
             raise ValueError(
                 f"start and end bounds ({len(start)}) must be the same length "
-                f"as the object ({num_vals})"
+                f"as the object ({num_vals}) divided by the step ({self.step}) "
+                f"if given and rounded up unless groupby was used"
```

Comment:
I think you can remove the groupby comment here since we're only scoping step for `rolling`